frost-core/src/frost.rs

```diff
@@ -396,8 +366,53 @@ where
         z = z + signature_share.signature.z_share;
     }
 
-    Ok(Signature {
+    let signature = Signature {
         R: group_commitment.0,
         z,
-    })
+    };
+
+    // Verify the aggregate signature
+    let verification_result = pubkeys
+        .group_public
+        .verify(signing_package.message(), &signature);
+
+    // If verification failed; verify each share to find the cheater.
```

Comment:
The "only if" construct might be handy here:
```suggestion
    // Only if the verification of the aggregate signature failed; verify each share to find the cheater.
```